crates/bevy_asset/src/assets.rs

```diff
@@ -177,7 +177,11 @@ impl<T: Asset> Assets<T> {
         mut events: EventWriter<AssetEvent<T>>,
         mut assets: ResMut<Assets<T>>,
     ) {
-        events.send_batch(assets.events.drain())
+        // Check if the events are empty before calling `drain`.
+        // As `drain` triggers change detection.
+        if !assets.events.is_empty() {
+            events.send_batch(assets.events.drain())
+        }
```

Comment:
I'm curious if people even think that this should possibly trigger change detection?
Users (AFAIK) don't have a way to access the underlying `event` of the `Assets` collection, and wouldn't be able to to look at the side effects of this. 
Though of course, it is possible the we could depend on this behavior internally....

Reply:
granted this would require a way to mutably access the contained value of `ResMut` without triggering change detection (which is something I can create a PR for later if desired).

Reply:
I'd prefer not to add this; interior mutability is enough of a confusing asetrix on our change detection already.

Reply:
That's fair enough :)

Reply:
Hmm yeah I think this might be a case where we want to ignore change detection. I don't think we use _any_ resource change detection internally, so we should be fine.

Reply:
I'll create a second PR (after this one) that introduces a way to to this and then applies it to this case.

Reply:
It'll be done via the new "DetectsChage" trait :)